Crash when pressing B button on playlists #16951

jbreitweiser opened the pull request:
## Description
 This is a correction to a null value when the full image name is not populated correctly.  Added check to avoid null pointer error when checking for alternative file names.

## Related Issues

Crash when pressing B button on playlists
#16951

## Reviewers

[If possible @mention all the people that should review your pull request]

zoltanvb commented:
Negation is missing (!string_is_empty), but unfortunately even after adding it, I can still reproduce the crash.
I suspect it is related to the loading procedure - the crash happens, when one moves away quickly from a playlist, I need to do left - backspace key sequence quickly one after the other, to make it happen reliably, though I have seen it happen in some other cases as well.

LibretroAdmin commented:
Maybe it's for the best if that PR that introduced all this code gets reverted and you go back to the drawing board on this stuff and make sure these crashes no longer occur on the next PR? That might be for the best vs. having to keep building on a broken foundation.

jbreitweiser commented:
@zoltanvb do you have icons in the logos thumbnail directory.  I want to
replicate your issue.  If so what are the file names.  I am going to back
track to the population of the path object to see if that img full name
variable in your stack trace might be an issue.  Can you also give me the
playlist you are using.  Maybe that will help replicate the issue.

On Sat, Sep 7, 2024, 9:04 AM LibretroAdmin ***@***.***> wrote:

> Maybe it's for the best if that PR that introduced all this code gets
> reverted and you go back to the drawing board on this stuff and make sure
> these crashes no longer occur on the next PR? That might be for the best
> vs. having to keep building on a broken foundation.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/libretro/RetroArch/pull/16971#issuecomment-2335180981>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BFFNNDCVOTRH63XQ5T7C6MDZVL2VRAVCNFSM6AAAAABNZWB3BWVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDGMZVGE4DAOJYGE>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

zoltanvb commented:
I do have a few for testing, but not for the last playlist where the crash usually happens, I believe they are not coming into play. Anyway I attach the playlist, with fake paths.
[Sinclair - ZX Spectrum.zip](https://github.com/user-attachments/files/16919082/Sinclair.-.ZX.Spectrum.zip)

Image full name and image short name can be empty on purpose, to save some queries, it has been like that since a while.
Short name: https://github.com/libretro/RetroArch/blob/c0715fc420d232b651e26a55861f2d93b63db67d/gfx/gfx_thumbnail_path.c#L53
Full name: https://github.com/libretro/RetroArch/blob/c0715fc420d232b651e26a55861f2d93b63db67d/gfx/gfx_thumbnail_path.c#L499

LibretroAdmin commented:
Alright @jbreitweiser , we'll give you the opportunity for now to track down this issue and resolve it. If that doesn't work we'll have to maybe consider the nuclear option of reverting it all, but for now you have the initiative.

jbreitweiser commented:
@zoltanvb I still cant replicate the issue but I think the issue is de initialization of the tumbnails.  I reverted the null checks.  The error seems to be a null pointer caused by moving off the playlist to a none playlistlist.  So I updated populate playlist (called when changing categories or lists) to check if there are active downloads and them de-initialize them.  Please see if you can still get the error.  Thanks.

zoltanvb commented:
@jbreitweiser I think push was missing, as I still see the previous commit.

jbreitweiser commented:
@zoltanvb pushed now.

On Sun, Sep 8, 2024, 1:27 PM zoltanvb ***@***.***> wrote:

> @jbreitweiser <https://github.com/jbreitweiser> I think push was missing,
> as I still see the previous commit.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/libretro/RetroArch/pull/16971#issuecomment-2336762778>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BFFNNDHDBCOCR67AWYR4OUDZVSCJHAVCNFSM6AAAAABNZWB3BWVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDGMZWG43DENZXHA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

zoltanvb commented:
Thank you! That does the trick, I can no longer reproduce the crash.

LibretroAdmin commented on menu/drivers/xmb.c:
```diff
@@ -2958,9 +2958,14 @@ static void xmb_populate_entries(void *data,
+         xmb_populate_dynamic_icons(xmb);
+   } 
+   else if(xmb->thumbnails.pending_icons != XMB_PENDING_THUMBNAIL_NONE )
+   {
```
See CODING-GUIDELINES, if a code block only has one line, then just drop the brackets

LibretroAdmin commented:
Other than that nitpick, it looks fine.

You'll have to rebase the PR though since some changes were made since your last push.

jbreitweiser commented:
code formatted per comments.
